src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstanceEventsController.cs

```diff
@@ -0,0 +1,177 @@
+        {
+            // or array of types is empty
+            if (string.IsNullOrEmpty(instanceId) || eventTypes.Count == 0 || eventTypes == null)
+            {
+                return BadRequest("Unable to perform query.");
+            }
+
+            List<InstanceEvent> result = await _repository.ListInstanceEventsSpecificEventTypes(instanceId, eventTypes);
+
+            if (result == null || result.Count == 0)
+            {
+                return NotFound($"Did not find any instance events for instanceId={instanceId} with event type {string.Join(", ", eventTypes.ToArray())}");
+            }
+
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Retrieves all instance events related to given instance id and time frame from instanceEvent collection.
+        /// </summary>
+        /// <param name="instanceId">Id of instance to retrieve events for.</param>
+        /// <param name="from"> Lower bound for DateTime span to filter events by. Utc format and invariantCulture. </param>
+        /// <param name="to"> Upper bound for DateTime span to filter events by. Utc format and invariantCulture. </param>
+        /// <returns>List of instance events.</returns>
+        /// GET storage/api/v1/instanceevents/GetByTimeFrame?instanceId={instanceId}&from={fromtime}&to={totime}
+        [HttpGet]
+        [Route("GetByTimeFrame")]
+        public async Task<ActionResult> GetInstanceEventsTimeframe(string instanceId, string from, string to)
+        {
+            DateTime fromDateTime, toDateTime;
+            try
+            {
+                fromDateTime = DateTime.ParseExact(from, "s", CultureInfo.InvariantCulture);
+                toDateTime = DateTime.ParseExact(to, "s", CultureInfo.InvariantCulture);
+            }
+            catch
+            {
+                return BadRequest("Unable to perform query.");
+            }
+
+            if (string.IsNullOrEmpty(instanceId))
+            {
+                return BadRequest("Unable to perform query.");
+            }
+
+            List<InstanceEvent> result = await _repository.ListInstanceEventsTimeFrame(instanceId, fromDateTime, toDateTime);
+
+            // should we differenciate between query failing and no instances existing in collection?
+            if (result == null || result.Count == 0)
+            {
+                return NotFound($"Did not find any instance events for instanceId={instanceId} between {fromDateTime} and {toDateTime}");
+            }
+
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Deletes all events related to an instance id.
+        /// </summary>
+        /// <param name="instanceId">Id of instance to retrieve events for. .</param>
+        /// <returns>Number of deleted events.</returns>
+        /// DELETE storage/api/v1/instanceevents/?instanceId={instanceId}
+        [HttpDelete]
+        public async Task<ActionResult> Delete(string instanceId)
```

Comment:
Here I will prefer to have an action on the events: e.g. instance-events/deleteAll?instanceId={instanceId}

Reply:
Would you still like this, @dskogan ? The routes have been changed now so the route for deleting is currently /instances/{instanceId}/events

Reply:
May be left as is when routes are updated.